### PR TITLE
Fix git annex: fix sha256 hash

### PR DESCRIPTION
**Pull request by changlinli**
It turns out that there's a latent bug where the git-annex hash is incorrect, but this will not trigger if you already have git-annex installed and are simply upgrading versions. See #25755 for more details.

In particular, the hash that is located in this file is for an old version of `git-annex` which is _not_ the one that is located in `hackage-packages.nix` which informs `drv.version`. This would cause `git-annex` to fail from a complete clean build (i.e. `git-annex` has never been built before).

Note that I'm also ignoring tests because somehow they can't find git-annex-shell while running. This does not seem indicative of an actual bug in the git-annex program itself and so instead of leaving git-annex in a broken state, I'm disabling tests for now.

###### Motivation for this change

Without this `git-annex` won't actually build on your machine if you're building it for the first time.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by changlinli** [comment]
I'm pretty sure the Travis failure has nothing to do with me and rather to do with the fact that somehow `haskell-tools-ast_0_6_0_0` has disappeared from `hackage-packages.nix`

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -62,17 +62,17 @@ self: super: {
 
   # The Hackage tarball is purposefully broken, because it's not intended to be, like, useful.
   # https://git-annex.branchable.com/bugs/bash_completion_file_is_missing_in_the_6.20160527_tarball_on_hackage/
-  git-annex = ((overrideCabal super.git-annex (drv: {
+  git-annex = dontCheck (((overrideCabal super.git-annex (drv: {
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: Please do not disable the tests. `git-annex` is used for manipulating important data and I don't want to distribute a version of the tool that does not pass it's own test suite.

**Review by peti — CHANGES_REQUESTED** [review]

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (changlinli) [inline pkgs/development/haskell-modules/configuration-common.nix]: Fair enough I'll try to give it a whack sometime today. The test suite isn't finding the git-annex-shell binary to test with

**Comment by peti** [comment]
> The test suite isn't finding the git-annex-shell binary to test with

I committed a fix for that issue already, but now the test suite fails
with different errors altogether. Not sure why that is, but I've
contacted the upstream author and asked for assistance.

**Comment by changlinli** [comment]
I suspect the tests have been broken (at least in nixpkgs) for several versions now. Reverting to 2017.03.01.1 still results in test failures. Indeed I'm puzzled this ever built in master in the first place now. Maybe some other consequences of #25755?

**Comment by changlinli** [comment]
Also I see you've manually modified the package in `hackage-packages.nix`. Is that okay? Because if it is, you could probably get rid of the problem around #25755 entirely by just modifying the `git-annex` entry there.

**Comment by peti** [comment]
The problem is git 2.13.0:

git bisect start
bad: [5d6c76dd50344753c611b0e73d705695e5c6cfca] git-annex: make sure the test suite runs with git-annex-shell in $PATH
git bisect bad 5d6c76dd50344753c611b0e73d705695e5c6cfca
good: [f2e100ca85994c6edab06d3b4aba612a8f7eba4c] Merge pull request #25616 from Ma27/package/activator/remove
git bisect good f2e100ca85994c6edab06d3b4aba612a8f7eba4c
bad: [18bad38d3d928b7dd9ee09a38249dbeb217d34d1] pythonPackages.callPackage to make it easier to call expressions that take many python packages
git bisect bad 18bad38d3d928b7dd9ee09a38249dbeb217d34d1
bad: [2dcc5bbb5d0aafd1b457e7decca67f5176fc177b] cmark: cmake in nativeBuildInputs
git bisect bad 2dcc5bbb5d0aafd1b457e7decca67f5176fc177b
bad: [2f6437393b33d6a32a2110d4e63f8420754c7a2f] Merge pull request #25669 from flosse/no-maintainer-anymore
git bisect bad 2f6437393b33d6a32a2110d4e63f8420754c7a2f
bad: [cb133234d7333868017ca9dc216b16556133e16d] Merge pull request #25664 from NeQuissimus/git_2_13_0
git bisect bad cb133234d7333868017ca9dc216b16556133e16d
good: [c6c7197c7dc5355ff587da9996330bb72afcb8f7] lnav: 0.8.1 -> 0.8.2
git bisect good c6c7197c7dc5355ff587da9996330bb72afcb8f7
good: [7b211da119589fa5d2889a0e399f208ba56149d3] Merge pull request #25531 from Infinisil/fix/ipfs-dataDir
git bisect good 7b211da119589fa5d2889a0e399f208ba56149d3
bad: [ce8662e693b5756e8457d94bb1765d853310afdb] git: 2.12.2 -> 2.13.0
git bisect bad ce8662e693b5756e8457d94bb1765d853310afdb
first bad commit: [ce8662e693b5756e8457d94bb1765d853310afdb] git: 2.12.2 -> 2.13.0

**Comment by peti** [comment]
@changlinli, no, modifying `hackage-packages.nix` manually is not okay and it's no solution to anything because the file will be overwritten by the automatic generator every few days. That's the very reason why that override for git-annex exists in the first place, because we cannot auto-generate a proper version.

**Comment by changlinli** [comment]
@peti Okay I thought as much. I was just confused by 5d6c76dd50344753c611b0e73d705695e5c6cfca. Maybe that was just a WIP commit?

**Comment by peti** [comment]
This issue is caused by a change in behavior in git: <https://git-annex.branchable.com/bugs/git-annex_in_nixpkgs_fails_with_git-2.13.0/>. Whether that change was intended by the git developers or not is not yet clear.

**Comment by peti** [comment]
The discussion takes place at <http://marc.info/?l=git&m=149495503107967&w=2>.

**Comment by joachifm** [comment]
This PR looks obsolete, is it?

**Comment by peti** [comment]
Fixed in `master`.

**Pull request closed without merging**